dask/dataframe/io/parquet/arrow.py

```diff
@@ -625,6 +654,181 @@ def _construct_parts(
+    partition_info = {
+        "partitions": partition_obj,
+        "partition_keys": partition_keys,
+        "partition_names": partition_names,
+    }
+    return metadata, partition_info
+
+
+def _gather_metadata_pyarrow_dataset(
+    paths, fs, split_row_groups, gather_statistics, filters, dataset_kwargs
+):
+    """ pyarrow.dataset version of _gather_metadata
+        Use pyarrow.dataset API to collect list of row-group fragments.
+        Also, collect other information necessary for parquet-to-ddf
+        mapping (e.g. schema, partition_info).
+    """
+    # Use pyarrow.dataset API
+    ds = None
+    if len(paths) == 1 and fs.isdir(paths[0]):
+
+        # Use _analyze_paths to avoid relative-path
+        # problems (see GH#5608)
+        base, fns = _analyze_paths(paths, fs)
+        paths = fs.sep.join([base, fns[0]])
+
+        meta_path = fs.sep.join([paths, "_metadata"])
+        if fs.exists(meta_path):
+            # Use _metadata file if available
+            ds = pa_ds.parquet_dataset(
+                meta_path, partitioning=dataset_kwargs.get("partitioning", "hive")
+            )
+            if gather_statistics is None:
+                gather_statistics = True
+    elif len(paths) > 1:
+        base, fns = _analyze_paths(paths, fs)
+        meta_path = fs.sep.join([base, "_metadata"])
+        if "_metadata" in fns:
+            # Use _metadata file if available
+            # Pyarrow cannot handle "_metadata" when `paths` is a list
+            paths.remove(meta_path)
+            fns.remove("_metadata")
+            ds = pa_ds.parquet_dataset(
+                meta_path, partitioning=dataset_kwargs.get("partitioning", "hive")
+            )
+            if gather_statistics is None:
+                gather_statistics = True
+
+    if ds is None:
+        ds = pa_ds.dataset(
+            paths,
+            format="parquet",
+            partitioning=dataset_kwargs.get(
+                "partitioning", "hive"
+            ),  # Assume "hive" by default
+        )
+
+    if split_row_groups is None and gather_statistics is not False:
+        split_row_groups = True
+
+    # Generate list of row-group fragments and call it `metadata`
+    metadata, partition_info = _collect_pyarrow_dataset_frags(ds, filters)
+    schema = ds.schema
+    base = ""
+
+    return (schema, metadata, base, partition_info, split_row_groups, gather_statistics)
+
+
+def _process_metadata_pyarrow_dataset(
+    metadata, single_rg_parts, gather_statistics, stat_col_indices, no_filters
+):
+
+    #
+    # TODO: Use `no_filters` to perform "early bail" opt.
+    #
+
+    # Get the number of row groups per file
+    file_row_groups = defaultdict(list)
+    file_row_group_stats = defaultdict(list)
+    file_row_group_column_stats = defaultdict(list)
+    for rg_frag in metadata:
+        row_group = rg_frag.row_groups[0]
+        fpath = rg_frag.path
+        file_row_groups[fpath].append(rg_frag)
+        statistics = row_group.statistics
```

Comment:
The `statistics` can be `None` here, if the dataset was not constructed from a `_metadata` file. 

I think there probably needs to be somewhere a 

```
if gather_statistics:
    rg_frag.ensure_complete_metadata()
```

Reply:
> Hmm - The statistics seems to be populated even when the dataset was not constructed from a _metadata file... Are you saying this behavior is not guaranteed, or am I just misinterpreting something?

Checking this: the full metadata (including statistics) gets parsed when a) constructing a dataset from a `_metadata` file, b) reading the full file, c) splitting by row groups with a filter, or d) when calling `ensure_complete_metadata()`.

So yes, `statistics` is not guaranteed to be populated in general (eg when not constructed from `_metadata` and not yet read in a first time and not splitted by row groups).  
Now, since it seems you are always calling `split_by_row_groups`, the statistics will indeed always be available, as you noticed. 

But I am wondering: is it needed to call `split_by_row_groups()` even when `split_row_groups` is not True?

> Note that I can add an ensure_complete_metadata call, but that seems to slow down _process_metadata_pyarrow_datasetby an order of magnitude. Does this make sense?

That should normally not happen. If the statistics were already parsed (eg by calling `split_by_row_groups()`), that method should be a no-op. 
So if that's indeed slowing down, that's something to investigate.

Reply:
> But I am wondering: is it needed to call split_by_row_groups() even when split_row_groups is not True?

Good point.  We should probably avoid doing this in most cases.  The exception is when we need statistics for index-column divisions.  In this case, we need to aggregate row-group-level min/max statistics to determine file-level min/max statistics.

I also remember there being limitations related to filtering at the file-fragment level (since filtering requires pyarrow to parse row-group statistics anyway).  If we avoid the row-group level fragment collection, will filters passed at the file level be applied correctly?

Reply:
> The exception is when we need statistics for index-column divisions. In this case, we need to aggregate row-group-level min/max statistics to determine file-level min/max statistics.

In theory, it should not be needed to split row groups just to get the statistics. For a non-splitted ParquetFragment, the `row_groups` property can give a list of RowGroupInfo objects with statistics

> I also remember there being limitations related to filtering at the file-fragment level (since filtering requires pyarrow to parse row-group statistics anyway). If we avoid the row-group level fragment collection, will filters passed at the file level be applied correctly?

Yes, that will still be done correctly, normally. It will not be filtered in advance, though, but only when actually scanning (reading) the file with `to_table` in the task. 
So thinking about it, that means that the divisions would be determined on statistics from non-filtered row groups .. (which I suppose deviates from the current behaviour?)

Reply:
So with a small example:

```
table = pa.table(pd.DataFrame(np.random.randn(100, 4), columns=list('ABCD')))    
pq.write_table(table, "test_row_groups.parquet", row_group_size=50)  
```

```
# using dataset with multiple fragments (with a single file it already seems to load the statistics from the beginning)
In [58]: dataset = ds.dataset(["test_row_groups.parquet", "test_row_groups.parquet"])       

In [59]: fragment = list(dataset.get_fragments())[0]  

In [60]: fragment.row_groups        

In [61]: fragment.ensure_complete_metadata()   

In [62]: fragment.row_groups                    
Out[62]: 
[<pyarrow._dataset.RowGroupInfo at 0x7f6761765818>,
 <pyarrow._dataset.RowGroupInfo at 0x7f6761765a28>]

In [63]: fragment.row_groups[0].statistics   
Out[63]: 
{'A': {'min': -1.685590262449956, 'max': 2.9169685774597043},
 'B': {'min': -2.5052142291876747, 'max': 1.9876548742106668},
 'C': {'min': -2.267851976381226, 'max': 2.9831708953348843},
 'D': {'min': -2.7986919637074514, 'max': 1.772406017664189}}
```

So you can see that you can get the statistics of all row groups, without explicitly splitting in multiple fragment object. 
But, as noted above, then don't use filtered row groups (but already filtered for potential partition fields) to determine divisions .. 
If there would be a need to somehow get a non-splitted-but-filtered ParquetFragment object, then we should think about an API for that / open an JIRA on the pyarrow side.

Reply:
> In theory, it should not be needed to split row groups just to get the statistics. For a non-splitted ParquetFragment, the row_groups property can give a list of RowGroupInfo objects with statistics

Great! Also, thank you for the useful example.  This should expand the scenarios in which we can avoid splitting on the dask end.

> So thinking about it, that means that the divisions would be determined on statistics from non-filtered row groups .. (which I suppose deviates from the current behaviour?)

Hmm that is a complication... Then we may need to require row-group splitting when a filter is involved (for now).